execution/block-partitioner/src/sharded_block_partitioner/messages.rs

```diff
@@ -1,84 +1,84 @@
 // Copyright © Aptos Foundation
 
 use crate::{
-    sharded_block_partitioner::dependency_analysis::{RWSet, WriteSetWithTxnIndex},
-    types::{SubBlock, TxnIndex},
+    sharded_block_partitioner::dependency_analysis::WriteSetWithTxnIndex,
+    types::{SubBlocksForShard, TxnIndex},
 };
 use aptos_types::transaction::analyzed_transaction::AnalyzedTransaction;
 use std::sync::Arc;
 
-pub enum ControlMsg {
-    DiscardCrossShardDepReq(DiscardTxnsWithCrossShardDep),
-    AddCrossShardDepReq(AddTxnsWithCrossShardDep),
-    Stop,
-}
-
-#[derive(Clone, Debug)]
-pub enum CrossShardMsg {
-    WriteSetWithTxnIndexMsg(WriteSetWithTxnIndex),
-    RWSetMsg(RWSet),
-    // Number of accepted transactions in the shard for the current round.
-    AcceptedTxnsMsg(usize),
-}
-
-pub struct DiscardTxnsWithCrossShardDep {
+pub struct DiscardCrossShardDep {
     pub transactions: Vec<AnalyzedTransaction>,
     // The frozen dependencies in previous chunks.
     pub prev_rounds_write_set_with_index: Arc<Vec<WriteSetWithTxnIndex>>,
-    pub prev_rounds_frozen_sub_blocks: Arc<Vec<SubBlock>>,
+    pub current_round_start_index: TxnIndex,
+    // This is the frozen sub block for the current shard and is passed because we want to modify
+    // it to add dependency back edges.
+    pub frozen_sub_blocks: SubBlocksForShard,
```

Comment:
It looks like the final sub-blocks won't be available until the last round. Maybe have every shard hold them and keep updating them every round? Passing them back and forth for every round seems inefficient.

Reply:
That's a good observation, I thought about it but didn't do it for 3 reasons 

1. Current design ensures the partitioning workers are stateless, which makes the design cleaner and also, going forward, ensures enhancements like adding fault tolerance/dynamic load balancing etc possible. I would like avoid making it a stateful design by storing the sub-blocks between rounds. 
2. Passing the block back and forth is minimal overhead as we pass ownership and don't do any cloning.
3. The partitioning APIs also is cleaner this way as otherwise, I would have to pass `SubBlocksForShard` only in first message and None in subsequent rounds. 

For above reasons, I would like to stick with current design.